buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestTestPlugin.java

```diff
@@ -0,0 +1,122 @@
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.bundling.Zip;
+
+/**
+ * Apply this plugin to run the YAML based REST tests.
+ */
+public class YamlRestTestPlugin implements Plugin<Project> {
+
+    public static final String SOURCE_SET_NAME = "yamlRestTest";
+
+    @Override
+    public void apply(Project project) {
+
+        // yaml Rest tests require a Java test runner
+        project.getPluginManager().apply(ElasticsearchJavaPlugin.class);
+        // to spin up the external cluster
+        project.getPluginManager().apply(TestClustersPlugin.class);
+        // to copy around the yaml tests and json spec
+        project.getPluginManager().apply(RestResourcesPlugin.class);
+
+        // note - source sets are not created via org.elasticsearch.gradle.util.GradleUtils.addTestSourceSet since unlike normal tests
+        // we only want the yamlRestTestSourceSet on the classpath by default. The yaml tests should be pure black box testing over HTTP and
+        // such it should not need the main class on the class path. Also, there are some special setup steps unique to YAML REST tests.
+
+        // create source set
+        SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+        SourceSet yamlTestSourceSet = sourceSets.create(SOURCE_SET_NAME);
+
+        // create task - note can not use .register due to the work in RestIntegTestTask's constructor :(
+        // see: https://github.com/elastic/elasticsearch/issues/47804
+        RestIntegTestTask yamlRestTestTask = project.getTasks()
+            .create(
+                SOURCE_SET_NAME,
+                RestIntegTestTask.class,
+                task -> { task.dependsOn(project.getTasks().getByName("copyRestApiSpecsTask")); }
+            );
+        yamlRestTestTask.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+        yamlRestTestTask.setDescription("Runs the YAML based REST tests against an external cluster");
+
+        // setup task dependency
+        if (BuildParams.isInternal()) {
+            project.getDependencies().add(SOURCE_SET_NAME + "Compile", project.project(":test:framework"));
+        } else {
+            project.getDependencies()
+                .add(SOURCE_SET_NAME + "Compile", "org.elasticsearch.test:framework:" + VersionProperties.getElasticsearch());
+        }
+
+        // setup the runner
+        RestTestRunnerTask runner = (RestTestRunnerTask) project.getTasks().getByName(yamlRestTestTask.getName() + "Runner");
+        runner.setTestClassesDirs(yamlTestSourceSet.getOutput().getClassesDirs());
+        runner.setClasspath(yamlTestSourceSet.getRuntimeClasspath());
+
+        // if this a module or plugin, it may have an associated zip file with it's contents, add that to the test cluster
+        boolean isModule = project.getPath().startsWith(":modules:");
+        Zip bundle = (Zip) project.getTasks().findByName("bundlePlugin");
+        if (bundle != null) {
```

Comment:
We should key this off the application of `PluginBuildPlugin`, not by a particular task. For [example](https://github.com/elastic/elasticsearch/blob/9526c7a4b340782892a2e4a57238f990c2ca2f42/buildSrc/src/main/java/org/elasticsearch/gradle/EnforceDeprecationFailuresPlugin.java#L47).

Reply:
Unfortunately, `PluginBuildPlugin` is in groovy, and this is in Java which can not reference the groovy side of things.

Reply:
No worries, we can do the same by referencing it by id.

```
project.getPluginManager().withPlugin("elasticsearch.esplugin", plugin -> {
    // do stuff
});
```

Reply:
done [`02afa30` (#56841)](https://github.com/elastic/elasticsearch/pull/56841/commits/02afa305711b7118a64ced93df1607097a5defcb) , thanks even think about using the ID.